src/app/services/xbitapi/xbit-api.service.ts

```diff
@@ -0,0 +1,395 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
+import { GenericHttpService } from '../generichttp/generic-http.service';
+import { getTypeNameForDebugging } from '@angular/core/src/change_detection/differs/iterable_differs';
+
+import { Address } from '../../entities/XBitApi/address';
+import { Administrator } from '../../entities/XBitApi/administrator';
+import { Algorithm } from '../../entities/XBitApi/algorithm';
+import { Balance } from '../../entities/XBitApi/balance';
+import { XBitCoin } from '../../entities/XBitApi/coin';
+import { Country } from '../../entities/XBitApi/country';
+import { Customer } from '../../entities/XBitApi/customer';
+import { Data } from '../../entities/XBitApi/data';
+import { HostingPeriod } from '../../entities/XBitApi/hostingperiod';
+import { Location } from '../../entities/XBitApi/location';
+import { Manufacturer } from '../../entities/XBitApi/manufacturer';
+import { Miner } from '../../entities/XBitApi/miner';
+import { MinerType } from '../../entities/XBitApi/minertype';
+import { MiningFarm } from '../../entities/XBitApi/miningfarm';
+import { UserInformation } from '../../entities/XBitApi/userinformation';
+
+@Injectable()
+
+export class XBitApiService {
+
+    baseUrl = "localhost:9999/api"
+
+    /********* Address *********/
+    // Get Address
+    public getAddress(id: string): Observable<Address> {
+        var url = this.baseUrl + '/address/' + id + '/';
+        return this.genericHttpService.getOne<Address>(url, Address);
+    }
+    // Post Address
+    public postAddress(body: any): Observable<any> {
+        var url = this.baseUrl + '/address/';
+        return this.genericHttpService.post(url, body);
+    }
+    // Put Address
+    public putAddress(body: any): Observable<any> {
+        var url = this.baseUrl + '/address/';
+        return this.genericHttpService.put(url, body);
+    }
+    // Delete Address
+    public deleteAddress(id: string): Observable<any> {
+        var url = this.baseUrl + '/address/' + id + '/';
+        return this.genericHttpService.delete(url);
+    }
+    /********* Administrator *********/
+    // Get Administrators
+    public getAdministrators(): Observable<Administrator[]> {
+        var url = this.baseUrl + '/administrator';
+        return this.genericHttpService.getMany<Administrator>(url, Administrator);
+    }
+    // Get Administrator
+    public getAdministrator(id: string): Observable<Administrator> {
+        var url = this.baseUrl + '/administrator/' + id + '/';
+        return this.genericHttpService.getOne<Administrator>(url, Administrator);
+    }
+    // Post Administrator
+    public PostAdministrator(body: any): Observable<any> {
```

Comment:
@XBITSwitzerland please stay consistent with function and variable names in pascal case